src/execution/execute.ts

```diff
@@ -1890,9 +1932,10 @@ function executeDeferredGroupedFieldSets(
   path: Path | undefined,
   newDeferredGroupedFieldSetRecords: ReadonlyArray<DeferredGroupedFieldSetRecord>,
   deferMap: ReadonlyMap<DeferUsage, DeferredFragmentRecord>,
+  parentIncrementalDataRecord: IncrementalDataRecord,
 ): void {
   for (const deferredGroupedFieldSetRecord of newDeferredGroupedFieldSetRecords) {
-    if (deferredGroupedFieldSetRecord.shouldInitiateDefer) {
+    if (!('deferredFragmentRecords' in parentIncrementalDataRecord)) {
```

Comment:
I think this new logic will never defer a new grouped field set if it was spawned by an existing deferred grouped field set, but I think we could have a situation where a defer spawns another defer in a child selection set and must actually be deferred, like:

```graphql
{
  ... @defer {
    field {
      ... @defer {
        subField
      }
    }
  }
}
```

Reply:
If you don't fall in this condition, `executeDeferredGroupedFieldSet` is still called, without being wrapped in `Promise.resolve().then(...)`.

Reply:
I'm not sure if this condition lines up exactly with `shouldInitiateDefer` in the current implementation

Reply:
@yaacovCR this is updated now and the logic should match the current implementation

Reply:
i will take another look